prow/spyglass/podlogartifact.go

```diff
@@ -71,23 +78,27 @@ func (a *PodLogArtifact) CanonicalLink() string {
 		"job": []string{a.name},
 		"id":  []string{a.buildID},
 	}
+	if a.artifactName != singleLogName {
+		q.Set("container", a.container)
+	}
```

Comment:
Shouldn't we also use the `buildLogRegex` here (or the `containerName` func) so that we can set the `container` query parameter in the multiple container case as well?

Reply:
This sets the container query parameter _only_ in the multiple container use case, otherwise it leaves the container query parameter as empty, in which case we automatically assume that the container name is kube.TestContainerName.  Regardless, this if statement seems unnecessary and I should remove it.